novawallet/Modules/NetworkDetails/NetworkDetailsInteractor.swift

```diff
@@ -166,6 +197,93 @@ private extension NetworkDetailsInteractor {
             nodesConnections[node.url] = connection
         }
     }
+
+    func disconnectNodes() {
+        nodesConnections.values.forEach { connection in
+            connection.disconnect(true)
+        }
+
+        nodesConnections = [:]
+    }
+
+    func measureNodePing(for nodeUrl: URL) {
+        guard
+            let connection = nodesConnections[nodeUrl.absoluteString],
+            let key = try? StorageKeyFactory().accountInfoKeyForId(
+                AccountId.zeroAccountId(of: chain.accountIdSize)
+            )
+        else {
+            return
+        }
+
+        let measureOperation: BaseOperation<Int> = if chain.isEthereumBased {
+            createPingMeasureOperation(
+                with: createEVMQueryOperation(with: connection),
+                queue: operationQueue
+            )
+        } else {
+            createPingMeasureOperation(
+                with: storageRequestFactory.queryOperation(
+                    for: { [key] },
+                    at: nil,
+                    engine: connection
+                ),
+                queue: operationQueue
+            )
+        }
+
+        execute(
+            operation: measureOperation,
+            inOperationQueue: operationQueue,
+            runningCallbackIn: .main
+        ) { [weak self] result in
+
+            switch result {
+            case let .success(ping):
+                self?.presenter?.didReceive(
+                    .pinged(ping),
+                    for: nodeUrl.absoluteString
+                )
+            case let .failure(error):
+                print(error)
+            }
+        }
+    }
+
+    func createEVMQueryOperation(with connection: ChainConnection) -> BaseOperation<Result<String, Error>> {
+        AsyncClosureOperation { resultClosure in
+            let params = EvmBalanceMessage.Params(
+                holder: String(),
+                block: .latest
+            )
+            _ = try connection.callMethod(
+                EvmBalanceMessage.method,
+                params: params,
+                options: .init(resendOnReconnect: false)
+            ) { (result: Result<String, Error>) in
+                resultClosure(.success(result))
+            }
+        }
+    }
+
+    func createPingMeasureOperation<T>(
+        with queryOperation: BaseOperation<T>,
+        queue: OperationQueue
+    ) -> BaseOperation<Int> {
+        AsyncClosureOperation { resultClosure in
+            let startTime = CFAbsoluteTimeGetCurrent()
+            execute(
+                operation: queryOperation,
+                inOperationQueue: queue,
+                runningCallbackIn: nil
+            ) { _ in
+                let endTime = CFAbsoluteTimeGetCurrent()
+                let ping = Int((endTime - startTime) * 1000)
+
+                resultClosure(.success(ping))
```

Comment:
I think we also need check the ping result and only provide measurement for the successful pings